src/index.js

```diff
@@ -40,25 +39,26 @@ function toMatchImageSnapshot(received, { customSnapshotIdentifier = '', customD
     updateSnapshot: snapshotState._updateSnapshot === 'all',
     customDiffConfig,
   });
+
   let pass = true;
+  let message = () => '';
+
   if (result.updated) {
     // once transition away from jasmine is done this will be a lot more elegant and pure
     // https://github.com/facebook/jest/pull/3668
     snapshotState = updateSnapshotState(snapshotState, { updated: snapshotState.updated += 1 });
   } else if (result.added) {
     snapshotState = updateSnapshotState(snapshotState, { added: snapshotState.added += 1 });
-    // see https://github.com/yahoo/blink-diff/blob/master/index.js#L251-L285 for result codes
-  } else if (result.code === 0 || result.code === 1) {
-    pass = false;
-  }
+  } else {
+    pass = result.pass;
 
-  // Clean up passing diff files
-  if (pass && fs.existsSync(result.diffOutputPath)) {
-    fs.unlinkSync(result.diffOutputPath);
-  }
+    if (!pass) {
+      const differencePercentage = parseInt(result.percentDiff * 100, 10);
```

Comment:
should be `result.diffPixels` no?

Reply:
Yep, i found that too. I also found a fairly significant bug in the integration test. I've added comments, hopefully it makes sense.